### PR TITLE
Reword the uploading pipelines documentation

### PR DESCRIPTION
**Pull request by toolmantim**
Ok, big updates to this doc. Can you review for sanity @keithpitt?

### CODE REVIEW

File: pages/agent/uploading_pipelines.md.erb [hunk 1]
```diff
@@ -1,42 +1,109 @@
 # Uploading Build Pipelines
 
-The Buildkite Agent’s `pipeline upload` command provides you with a way of storing build pipelines within source control, and have them uploaded to builds at runtime.
+In addition to defining your build pipelines via the web you can use your buildkite agent’s `pipeline upload` command to read them from your source code, or even dynamically generate them using a script. This allows you to version, audit and review your build pipelines alongside your source code.
```
Comment (keithpitt) [inline pages/agent/uploading_pipelines.md.erb]: `buildkite` here should be capitalised

File: pages/agent/uploading_pipelines.md.erb [hunk 2]
```diff
@@ -1,42 +1,109 @@
...
-  buildkite-agent pipeline upload --replace
-else
-  echo "No .buildkite/pipeline.yml file, using existing steps"
-fi
+A *blocker* step will pause the pipeline and wait for a team member to unblock it (via the web or the API).
+
+```yml
+steps:
+  - command: command.sh
+  - block
```
Comment (keithpitt) [inline pages/agent/uploading_pipelines.md.erb]: This should be:

```
block:
  name: ":rocket: Deploy"
```

**Comment by keithpitt** [comment]
Nice!

**Comment by keithpitt** [comment]
:dancers:

**Comment by toolmantim** [comment]
Awesome, thanks for the catches!!

**Pull request merged**